### DISCUSSION TITLE
Datenbank Spieler

### DISCUSSION THREAD

**Discussion by schimmi70** [obermacker/spacefights#6] [General]
Ich hab mal drüber geschaut, es werden beide Tabellen gebraucht.
Folgende Punkte sollten wir besprechen:

- wir sollten uns einig werden ob "spieler" nun "player" oder "gamer" heissen soll.
- wir sollten die Galaxy-spieler Tabelle in die Spieler-Datenbank überführen (da würde ich sie suchen) 
- wir sollten möglichst doppelte Einträge vermeinden und über die SpielerID joinen

Ich würde wenn du nichts dagegen hast einen Entwurf machen  und den hier zum anschauen reinsetzen.

VG Alban

**Comment by obermacker** [comment]
Die Datenbank [Spieler] dient dem Login in den Auswahlbereich der Galaxy.

`ID`			Primärschlüssel
`spieler_ID`		Eindeutige ID wird in die Galaxy vererbt
`spieler_name` 		Accountname und soll den Namen innerhalb der Galaxy schützen (_zu testen_)
`name_galaxy_1` 	Spielername in Galaxy 1
`name_galaxy_2` 	Spielername in Galaxy 2
`name_galaxy_3` 	Spielername in Galaxy 3
`passwort` 		Passwort verschlüsselt
`letzter_login` 		Letzter Login (_um ungenutzte Accounts bereinigen zu können_)
`session_id` 		Session ID (_Beschreibung hinzufügen_)
`HTTP_USER_AGENT` 	um parallele Logins zu verhindern (_naja_)
`spieler_geloescht` 	markiert wenn ein Spieler sich löscht (_nicht implementiert_)
`name_gala_1` 		_ungenutzt/doppelt?_
`aktiv_gala_1` 		_ungenutzt/doppelt?_
`letzter_Planet` 	**gehört hier nicht hin**. Speichert den zuletzt ausgewählten Planeten über den Login hinweg innerhalb einer Galaxy.  (_gehört in die Galaxy_)
`max_Planeten` 	**gehört hier nicht hin**. Speichert die Anzahl Kolonien des Spielers innerhalb einer Galaxy.  (_gehört in die Galaxy_)
`chat_color`		**gehört hier nicht hin**. Speicher die Chatfarbe des Spielers. (_gehört in die Galaxy_)

Diese Datenbank soll nur für den Login elementare Daten speichern und zusätzlich der Sicherheit des Spielbetriebes dienen.
Sollten sich Datenbankanpassungen in einer einzelnen Galaxy ergeben so bleibt diese Datenbank unangetastet und läuft nicht Gefahr kaputt zu gehen.

Die `Spieler` Tabelle innerhalb einer Galaxy dient dem Betrieb eben dieser und kann, ohne die generelle Accountsicherheit zu gefährden angepasst werden. Im schlimmsten Falle sperrt man den/die Spieler nur von der Galaxy aus, aber nicht vom eigentlichen Spiel.

Beispiel:
-	Spieler A loggt sich ein und steht im Auswahlbildschirm der Galaxy 1 – 3. 
o	Galaxy 1 (_stable_)
o	Galaxy 2 (_speed_ oder _season_)
o	Galaxy 3 (_unstable_)
-	Dann ein Problem:
o	Galaxy 3 erfährt einen Patch und läuft in einen schweren Fehler, so dass kein Spieler in dieser Galaxy vernünftig spielen kann
o	Zeitgleich in Galaxy 2 hängt Spieler A im Gebäudebau fest und kann keine weiteren Gebäude bauen, weil das Skript in einem Zählerüberlauf festhängt. 
-	Spieler A kann sich also immer noch einloggen und
o	in Galaxy 1 ohne Probleme weiterspielen
o	in Galaxy 2 auf einen fix warten
o	in Galaxy 3 auch auf den Fix warten, aber da die auf _unstable_ läuft ist das unkritisch
-	Zu keiner Zeit waren die Logindaten der Datenbank [Spieler] in Gefahr.

**Reply by schimmi70** [reply]
Warum es Zwei sind habe ich auch erst nach meinem Post realiesiert ;)
Klinkt soweit auch alles logisch, sollten wir die erstmal zeitnah bereinigen ?

**Comment by obermacker** [comment]
>     * wir sollten uns einig werden ob "spieler" nun "player" oder "gamer" heissen soll.

Ich habe Deine Idee aufgegriffen und hier als Thread gestartet [https://github.com/obermacker/spacefights/discussions/9#discussion-6164604](https://github.com/obermacker/spacefights/discussions/9#discussion-6164604)
 
>     * wir sollten möglichst doppelte Einträge vermeinden und über die SpielerID joinen

Wie meinst Du das?

VG Steve

**Comment by schimmi70** [comment]
In den Galaxy-Tabellen  wiederholen sich viele Angaben, wäre es da nicht einfacher z.B. nur die SpielerID in die verschiedenen Tabellen schreiben und den Namen immer über die Tabelle Spieler hinzufügen. Entweder mit eigenen Aufruf oder bei select from noch ein "join" Statement nutzen und die Daten aus zwei Tabellen ziehen. Zeitlich (auführungstechnisch) sollte das kein Problem sein.

**Comment by obermacker** [comment]
Die Spieler/Planetennamen in Tabelle `Flotten` z.B. sind da nur drin, weil es sich für mich einfacher debuggen ließ. So braucht man einfach in die DB schauen um zu wissen was gerade los ist. 

Da hast Du recht, solche Sachen können vemieden werden.
